src/daemon/auth/providers/openid_connect/crypt.rs

```diff
@@ -0,0 +1,34 @@
+// TOOD: use the ring::aead functionality instead of the chacha20_poly1305_aead
+// crate, as we already depend on ring.
+
+// See: https://tools.ietf.org/html/rfc7539#section-2.8
+//      https://briansmith.org/rustdoc/ring/aead/index.html
+//      http://www-cse.ucsd.edu/~mihir/papers/oem.html
+
+use std::io::Write;
+
+use crate::commons::error::Error as KrillError;
+use crate::commons::KrillResult;
+
+// TODO: use proper values
```

Comment:
Don't even hard-code them in the source code, the key needs to be generated and stored on disk or something.